src/main/java/uk/ac/manchester/tornado/drivers/spirv/levelzero/LevelZeroPowerMonitor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package uk.ac.manchester.tornado.drivers.spirv.levelzero;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LevelZeroPowerMonitor {
+
+    public native long[] getSysmanDevicesToQuery(long[] allSysmanDevices);
+
+    // public native List<ZesPowerEnergyCounter> getEnergyCounters(long sysmanDeviceHandle);
+
+    private native int zesDeviceEnumPowerDomains_native(long deviceHandler, int[] numPowerDomains, long[] hMemory);
+
+    public int zesDeviceEnumPowerDomains(long deviceHandler, int[] numPowerDomains, long[] hMemory) {
+        int result = zesDeviceEnumPowerDomains_native(deviceHandler, numPowerDomains, hMemory);
+        return result;
+    }
+
+    public native int getEnergyCounters_native(long sysmanDeviceHandle, List<ZesPowerEnergyCounter> energyCounterList);
+    
+    public int getEnergyCounters(long sysmanDeviceHandle, List<ZesPowerEnergyCounter> energyCounterList) {
+        
+        int result = getEnergyCounters_native(sysmanDeviceHandle, energyCounterList);
+        return result;
+        
+    }
+
+    public PowerQueryStatus queryBasedOnPowerDomains(long sysmanDevice) {
```

Comment:
I think this method should be refactored to a name like:
`getPowerSupportStatusForDevice`.

Reply:
Also, this method is used twice. I suggest to make a second method that has no arguments and just checks if there are any power domains in the system, with name `isPowerDomainPresent()`. It can return a boolean type or the PowerQueryStatus.